### PR TITLE
fix: ignore unknown session end events

### LINKED ISSUE #1251: session/end creates malformed row when session does not exist
## Summary

`POST /agentmemory/session/end` calls `state::update` without first checking whether the session exists. Because iii-engine's update operation has upsert semantics, ending an unknown session creates a malformed row containing only `status` and `endedAt`.

That row has no `id`, so `GET /agentmemory/sessions` later calls `kv.get(KV.summaries, undefined)`. The state invocation can remain pending until timeout, making the Viewer and MCP clients appear to have zero sessions even though the original sessions are still present.

## Reproduction

1. Run agentmemory 0.9.29 with the OpenClaw integration.
2. Let OpenClaw have a session that has not yet produced a successful `agent_end` observation.
3. Send `/new`. OpenClaw correctly emits `session_end(reason: "new")`.
4. The plugin posts that logical session key to `/agentmemory/session/end`.
5. Request `GET /agentmemory/sessions`.

The issue is also reproducible directly by posting any nonexistent ID:

```bash
curl -X POST http://localhost:3111/agentmemory/session/end \
  -H 'content-type: application/json' \
  -d '{"sessionId":"unknown-session"}'
```

## Actual result

The sessions state gains a partial value similar to:

```json
{
  "status": "completed",
  "endedAt": "2026-08-26T08:26:18.409Z"
}
```

Subsequent `/agentmemory/sessions` requests can time out while resolving a summary with an undefined key. The dashboard then looks as if all sessions were cleared. Inspecting `state::list` confirms the valid sessions were never deleted.

## Expected result

- Ending an unknown session should be an idempotent no-op.
- It must not create a session row.
- The stopped lifecycle should only run when an existing active session is actually transitioned to completed.
- Session-listing code should tolerate historical malformed rows rather than allowing one row to fail the whole response.

## Proposed fix

1. Under the same per-session lock used by observation capture, read the session before updating it.
2. Return a successful no-op for missing, malformed, or already-completed sessions.
3. Only publish `event::session::stopped` after a real state transition.
4. Filter rows without a non-empty string `id` before joining summaries in `api::sessions`.
5. Add behavioral regression tests for the missing-session and malformed-row cases.

## Related issues

- #1058 — OpenClaw lifecycle integration and `session_end`
- #466 — Viewer failure caused by sessions missing `id`/`project`
- #729 — another reader poisoned by records missing `id`
- #1100 — same `/sessions` timeout/empty-dashboard symptom, but a different fan-out root cause

## Environment

- agentmemory 0.9.29
- iii-engine 0.11.2
- OpenClaw 2026.7.1
- macOS, Node.js 25.9.0

### PR DESCRIPTION
**Pull request by a652**
## Summary

- make `POST /agentmemory/session/end` an idempotent no-op for missing, malformed, or already-completed sessions
- serialize session-end updates with observation capture using the existing per-session lock
- only publish `event::session::stopped` after a real active-to-completed transition
- skip legacy session rows without a valid `id` before looking up summaries
- add behavioral regression coverage for unknown sessions, partial rows, valid session completion, and resilient session listing

## Root cause

iii-engine's `state::update` has upsert semantics. Calling `/agentmemory/session/end` for a session that had not yet been created therefore wrote a partial row containing only `status` and `endedAt`.

`api::sessions` later attempted `kv.get(KV.summaries, s.id)` for that row, where `s.id` was undefined. That invocation could remain pending until timeout, making the dashboard and MCP surface appear to contain zero sessions even though the valid rows were still present.

OpenClaw exposes this race when `/new` emits `session_end(reason: "new")` before the conversation has produced its first successful observation, but the server endpoint should be safe for every integration.

## Verification

- `npm test`: 1,715 passed, 1 skipped
- `npm run build`: passed
- live regression against iii-engine 0.11.2:
  - ending an unknown session returns HTTP 200 with `ended: false`
  - no session row is created
  - `/agentmemory/sessions` continues to return all valid sessions

Fixes #1251

Related: #1058, #466, #729, #1100

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved session ending behavior for unknown, incomplete, or already completed sessions.
  - Prevented malformed session records from appearing in session listings.
  - Ensured session completion events are published only once.
  - Prevented session summaries from being requested for invalid records.

- **Tests**
  - Added coverage for session lifecycle and listing edge cases.

### CODE REVIEW

File: test/session-lifecycle-api.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import { describe, expect, it, vi } from "vitest";
+
+import { registerApiTriggers } from "../src/triggers/api.js";
+import { KV } from "../src/state/schema.js";
+import type { Session } from "../src/types.js";
```
[the remark this answers is not in the document]
Comment (a652) [reply test/session-lifecycle-api.test.ts:16]: Added the required `vi.mock("iii-sdk")` setup while keeping the injected `sdk.trigger` and `kv.get`/`set`/`list` mocks in the harness. Full suite passes (1716 passed, 1 skipped). Addressed in 5f5a949.

File: test/session-lifecycle-api.test.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    const response = (await handler({ body: { sessionId } })) as ApiResponse;
+
+    expect(response.body).toEqual({
+      success: true,
+      ended: false,
+      reason: "not_found",
+    });
+    expect(harness.update).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (a652) [reply test/session-lifecycle-api.test.ts:122]: Added a valid completed-session regression case covering `already_completed`, with assertions that neither `kv.update` nor the stopped lifecycle trigger runs. Addressed in 5f5a949.
